ADRs/0004-separate-consensus-and-p2p-keys.md

```diff
@@ -0,0 +1,52 @@
+# Separate Consensus and P2P Private Keys
+
+* Status: in review
+* Deciders: Pocket Network Team
+* Date: 2023-04-17
+
+Technical Story: [Consolidate and Refactor Node Identity](https://github.com/pokt-network/pocket/issues/348)
+
+## Context and Problem Statement
+
+In the context of maintaining a secure and robust network, we need to decide whether to use separate private keys for P2P and consensus modules, or use a single private key for both purposes.
+Combining the keys could simplify the identity management, but may also increase security risks and impact system quality.
+
+## Decision Drivers
+
+* Security: Minimizing the risks associated with key compromise
+* Simplification: Reducing complexity in identity management
+* Flexibility: Allowing different key management strategies for different modules
+* Isolation: Minimizing the impact of compromise on other system components
+
+## Considered Options
+
+* Option 1: Use a single private key for both P2P and consensus modules
+* Option 2: Use separate private keys for P2P and consensus modules
+
+## Decision Outcome
+
+Chosen option: "Option 2: Use separate private keys for P2P and consensus modules", because this approach provides a better security posture, enables greater flexibility in key management strategies, and isolates the impact of potential key compromises.
```

Comment:
As I'm reading this, I was thinking about `Utility` as well.

P2P - I'm thinking TLS key rotation, gossip, etc...
Consensus - I'm thinking validators but also Utility related logic (e.g. staking / unstaking / sending messages)

Do you think it's fair to change the ADR to separate keys between P2P and "everything else"? Specifically, having separate keys for consensus and utility seems like a wrong decision, which is why I'm proposing this change.